### PR TITLE
Cassandra 4.1

### PR DESCRIPTION
**Pull request by bschoening**
Update the code to match the `cassandra` `4.1` release tag:
* https://cassandra.apache.org/_/blog/Cassandra-4.1-is-here.html
* https://github.com/apache/cassandra/releases/tag/cassandra-4.1.0

There are a number of `cqlsh` changes, grep the changelog for `cqlsh` to see details:
* https://github.com/apache/cassandra/blob/cassandra-4.1.0/CHANGES.txt

### CODE REVIEW

File: cqlsh/cqlsh.py [hunk 1]
```diff
@@ -116,20 +112,14 @@ def find_zip(libprefix):
     ver = os.path.splitext(os.path.basename(cql_zip))[0][len(CQL_LIB_PREFIX):]
     sys.path.insert(0, os.path.join(cql_zip, 'cassandra-driver-' + ver))
 
-third_parties = ('futures-', 'six-', 'geomet-')
+# the driver needs dependencies
+third_parties = ('six-', 'pure_sasl-')
```
Comment (jeffwidman) [inline cqlsh/cqlsh.py:116]: Unrelated to this PR: The driver hasn't dropped `six` yet? Weird...

**Review by jeffwidman — APPROVED** [review]
I assume this is basically a direct copy from the `cassandra` `4.1` tag? I didn't double-check...

Thanks for all your hard work upstream to improve `cqlsh`, I know you've put a lot of time into that, and it shows in these improvements and dropping all kinds of deprecations.

Everything here looks good except the PR description, so I'll tweak that quickly here.

File: cqlsh/cqlsh.py (hunk 1 above)
Comment (jeffwidman) [reply cqlsh/cqlsh.py:116]: Actually, grep'ing the `cqlshlib` code, it looks like `six` was removed everywhere... 

I assume this code still matches upstream, in which case we should keep it, but maybe upstream should be updated to drop `six-` from `third_parties`? I dunno, I haven't actually looked at this very deeply so I may be missing something...

**Comment by bschoening** [comment]
Yes, it’s a direct copy from 4.1.

On trunk is my refactoring of csqlsh.py, that will be nice to see at some point in the future.

The python driver seems to have been abandoned by DataStax and hasn’t been updated in quite a while.

**Comment by bschoening** [comment]
No, that’s a gotcha, because the driver still uses it, it needs to be dynamically loaded, even though cqlshlib doesn’t use it.

**Pull request merged**